### PR TITLE
Optimize read accessors to just borrow yielded storage references

### PR DESCRIPTION
**Pull request by rjmccall**

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please test.

**Comment by harlanhaskins** [comment]
Are yield lists comma separated? If so, you’ll probably need to add a production that includes an optional trailing comma, like the definition for TupleElement.

Also, can you add a round-trip test to test/Syntax/round_trip_parse_gen.swift?

**Comment by rjmccall** [comment]
@swift-ci Please test.

**Comment by rjmccall** [comment]
> Are yield lists comma separated? If so, you’ll probably need to add a production that includes an optional trailing comma, like the definition for TupleElement.

The parser doesn't actually accept an optional trailing comma currently.

> Also, can you add a round-trip test to test/Syntax/round_trip_parse_gen.swift?

Done.

File: lib/SILGen/SILGenApply.cpp [hunk 1]
```diff
@@ -2773,6 +2887,64 @@ class ArgEmitter {
...
+                       SILType loweredSubstParamType,
+                       AbstractionPattern origParamType,
+                       ClaimedParamsRef paramsSlice) {
+    assert(paramsSlice.size() == 1);
+
+    // Try to find an expression we can emit as an l-value.
+    if (!arg.isExpr()) return false;
+    auto argExpr = arg.peekExpr();
+    auto lvExpr = findStorageReferenceExprForBorrow(argExpr);
+    if (!lvExpr) return false;
```
[the remark this answers is not in the document]
Comment (rjmccall) [reply lib/SILGen/SILGenApply.cpp]: Good point; I'll do a follow-up if this test run goes through.

**Pull request merged**